stable/jenkins/templates/config.yaml

```diff
@@ -66,17 +67,29 @@ data:
           <serverUrl>https://kubernetes.default</serverUrl>
           <skipTlsVerify>false</skipTlsVerify>
           <namespace>{{ template "jenkins.master.slaveKubernetesNamespace" . }}</namespace>
-{{- if .Values.master.slaveKubernetesNamespace }}
-          <jenkinsUrl>http://{{ template "jenkins.fullname" . }}.{{ template "jenkins.namespace" . }}:{{.Values.master.servicePort}}{{ default "" .Values.master.jenkinsUriPrefix }}</jenkinsUrl>
-          <jenkinsTunnel>{{ template "jenkins.fullname" . }}-agent.{{ template "jenkins.namespace" . }}:{{ .Values.master.slaveListenerPort }}</jenkinsTunnel>
+{{- if .Values.master.slaveJenkinsUrl }}
+          <jenkinsUrl>{{ .Values.master.slaveJenkinsUrl }}</jenkinsUrl>
 {{- else }}
+  {{- if .Values.master.slaveKubernetesNamespace }}
+          <jenkinsUrl>http://{{ template "jenkins.fullname" . }}.{{ template "jenkins.namespace" . }}:{{.Values.master.servicePort}}{{ default "" .Values.master.jenkinsUriPrefix }}</jenkinsUrl>
+  {{- else }}
           <jenkinsUrl>http://{{ template "jenkins.fullname" . }}:{{.Values.master.servicePort}}{{ default "" .Values.master.jenkinsUriPrefix }}</jenkinsUrl>
+  {{- end }}
+{{- end }}
+
+{{- if .Values.master.slaveJenkinsTunnel }}
+          <jenkinsTunnel>{{ .Values.master.slaveJenkinsTunnel }}</jenkinsTunnel>
+{{- else }}
+  {{- if .Values.master.slaveKubernetesNamespace }}
+          <jenkinsTunnel>{{ template "jenkins.fullname" . }}-agent.{{ template "jenkins.namespace" . }}:{{ .Values.master.slaveListenerPort }}</jenkinsTunnel>
+  {{- else }}
           <jenkinsTunnel>{{ template "jenkins.fullname" . }}-agent:{{ .Values.master.slaveListenerPort }}</jenkinsTunnel>
+  {{- end }}
 {{- end }}
           <containerCap>{{ .Values.agent.containerCap }}</containerCap>
           <retentionTimeout>5</retentionTimeout>
-          <connectTimeout>0</connectTimeout>
-          <readTimeout>0</readTimeout>
+          <connectTimeout>{{ .Values.master.slaveConnectTimeout }}</connectTimeout>
+          <readTimeout>{{ .Values.master.slaveReadTimeout }}</readTimeout>
```

Comment:
please also add `master.slaveReadTimeout`to the JCasC as `readTimeout` [here](https://github.com/helm/charts/blob/master/stable/jenkins/templates/_helpers.tpl#L101)